File: 26_string_manipulation/26_1_string_split.cpp
#include <iostream>
#include <vector>
#include <string>

std::vector<std::string> split(const std::string& s, char c) {
  std::vector<std::string> output;
  std::string acc = "";

  for (char el : s) {
    if (el == c) {
      output.push_back(acc);
      acc = "";
    } else {
      acc += el;
    }
  }

  output.push_back(acc);

  return output;
}

int main() {
  std::string s1 = "split by space";
  char c1 = ' ';
  std::vector<std::string> result1 = split(s1, c1);
  std::cout << "[";
  for (size_t i = 0; i < result1.size(); i++) {
    std::cout << "\"" << result1[i] << "\"";
    if (i < result1.size() - 1) std::cout << ", ";
  }
  std::cout << "]" << std::endl;
}

File: 26_string_manipulation/26_1_string_split.py
def split(s: str, c: str) -> list:
  output = []
  acc = []
  for el in s:
    if el == c:
      output.append("".join(acc))
      acc = []
    else:
      acc.append(el)

  output.append("".join(acc))
  
  return output

if __name__ == "__main__":
  # Example1
  s = "split by space" 
  c = " "
  print(split(s, c))
  # Example2
  s = "beekeeper needed"
  c = "e"
  print(split(s, c))
  # Example3
  s = "/home/./..//Documents/"
  c = "/"
  print(split(s, c))